Sources/Soto/Extensions/S3/S3+multipart.swift

```diff
@@ -34,6 +34,24 @@ extension S3 {
     public enum ThreadPoolProvider {
         case createNew
         case shared(NIOThreadPool)
+
+        func create() -> NIOThreadPool {
+            let threadPool: NIOThreadPool
+            switch self {
+            case .createNew:
+                threadPool = NIOThreadPool(numberOfThreads: NonBlockingFileIO.defaultThreadPoolSize)
+                threadPool.start()
+                return threadPool
+            case .shared(let sharedPool):
+                return sharedPool
+            }
+        }
+
+        func destory(_ threadPool: NIOThreadPool) {
+            if case .createNew = self {
+                threadPool.shutdownGracefully { _ in }
```

Comment:
Would it make sense to use `syncShutdownGracefully` here to account for the async nature of it? I'm aware this then throws etc and can ripple everywhere

Reply:
This is used by both async and non-async code. Just noticed a typo though `destroy` not `destory`.
I guess it would be useful if there was an async version of this. I could easily add one.
```swift
extension S3.ThreadPoolProvider {
    func destroy(_ threadPool: NIOThreadPool) async throws {
        if case .createNew = self {
            return try await withUnsafeThrowingContinuation { cont in
                threadPool.shutdownGracefully { error in
                    if let error = error {
                        cont.resume(throwing: error)
                    } else {
                        cont.resume()
                    }
                }
            }
        }
    }
}
```